src/main/java/org/apache/mesos/scheduler/repair/CoreFrameworkState.java

```diff
@@ -0,0 +1,44 @@
+package org.apache.mesos.scheduler.repair;
+
+import org.apache.mesos.Protos.FrameworkID;
+import org.apache.mesos.Protos.TaskID;
+import org.apache.mesos.Protos.TaskInfo;
+import org.apache.mesos.state.StateStoreException;
+
+import java.util.List;
+
+/**
+ * Exposes simple task-oriented API for components that need to interact with the framework
+ * as a whole.
+ */
+public interface CoreFrameworkState {
+    /**
+     * Get the {@link FrameworkID} for this framework.
+     *
+     * @return
+     */
+    FrameworkID getFrameworkId();
+
+    /**
+     * Get all the {@link TaskInfo}s for this framework.
+     * @return
+     * @throws StateStoreException
+     */
+    List<TaskInfo> getTaskInfos() throws StateStoreException;
+
+    /**
+     * Get the {@link TaskInfo}s of all tasks in terminal states for this framework.
+     * @return
+     * @throws StateStoreException
+     */
+    List<TaskInfo> getTerminatedTaskInfos() throws Exception;
+
+    /**
+     * Remove the given task from this framework. This could cause the given task to be killed
+     * if it's still running, and it will remove the task from the framework's tracking.
+     * The absence of the task will be fixed by the {@link RepairScheduler}, if that task
+     * is still supposed to be running.
+     * @param taskId The ID of the task to delete
+     */
+    void deleteTask(TaskID taskId);
```

Comment:
As, stated previously.  Deletion of a Task is an implementation detail of how Kafka works, not how all users of Auto Recovery should work.  I don't think this method belongs here at all, and it's meaning is difficult to understand.  `This could cause the given task to be killed`...  It's unclear what happens when this method is called.  It might kill something, but it will definitely delete something from persistent storage?